### PR TITLE
make sure to call clean inside overwriteFileJS

### PR DESCRIPTION
**Pull request by kevodwyer**
web-ui PR: https://github.com/Peergos/web-ui/pull/166

### CODE REVIEW

File: src/peergos/server/simulation/PeergosFileSystemImpl.java [hunk 1]
```diff
@@ -55,6 +55,14 @@ public void write(Path path, byte[] data, Consumer<Long> progressConsumer) {
 
     }
 
+    @Override
+    public void modify(Path path, byte[] data, Consumer<Long> progressConsumer) {
+        FileWrapper file = getPath(path);
+        ProgressConsumer<Long> pc  = l -> progressConsumer.accept(l);
```
Comment (ianopolous) [inline src/peergos/server/simulation/PeergosFileSystemImpl.java:61]: can just use progressConsumer directly

File: src/peergos/shared/user/fs/FileWrapper.java [hunk 2]
```diff
@@ -588,11 +588,12 @@ public static int getNumberOfChunks(long size) {
...
         long newSize = LongUtil.intsToLong(endHigh, endLow);
         return network.synchronizer.applyComplexUpdate(owner(), signingPair(),
-                (s, committer) -> overwriteSection(s, committer, fileData,
+                (s, committer) -> clean(s, committer, network, crypto)
+                        .thenCompose(u -> u.left.overwriteSection(u.right, committer, fileData,
                         0L, newSize, network, crypto, monitor)
                         .thenCompose(v -> newSize >= size ? Futures.of(v) :
                                 getUpdated(v, network)
-                                        .thenCompose(f -> f.truncate(v, committer, newSize, network, crypto))))
+                                        .thenCompose(f -> f.truncate(v, committer, newSize, network, crypto)))))
```
Comment (ianopolous) [inline src/peergos/shared/user/fs/FileWrapper.java:596]: We can close the scope as soon as we are done with it to reduce indentation hell. here, line 593.

File: src/peergos/server/simulation/PeergosFileSystemImpl.java (hunk 1 above)
Comment (kevodwyer) [reply src/peergos/server/simulation/PeergosFileSystemImpl.java:61]: Copied from another method. I have inlined

File: src/peergos/shared/user/fs/FileWrapper.java (hunk 2 above)
Comment (kevodwyer) [reply src/peergos/shared/user/fs/FileWrapper.java:596]: re-formatted with a little help from IDE

File: src/peergos/shared/user/fs/FileWrapper.java [hunk 3]
```diff
@@ -588,12 +588,12 @@ public static int getNumberOfChunks(long size) {
         long size = getSize();
         long newSize = LongUtil.intsToLong(endHigh, endLow);
         return network.synchronizer.applyComplexUpdate(owner(), signingPair(),
-                (s, committer) -> overwriteSection(s, committer, fileData,
-                        0L, newSize, network, crypto, monitor)
+                (s, committer) -> clean(s, committer, network, crypto)
+                    .thenCompose(u -> u.left.overwriteSection(u.right, committer, fileData,
+                            0L, newSize, network, crypto, monitor)
```
Comment (ianopolous) [inline src/peergos/shared/user/fs/FileWrapper.java:593]: Still not closing the scope early. line 593 should end the clean thencompose

Comment (kevodwyer) [reply src/peergos/shared/user/fs/FileWrapper.java]: fixed

**Pull request merged**